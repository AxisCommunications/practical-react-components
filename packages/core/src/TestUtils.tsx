import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
