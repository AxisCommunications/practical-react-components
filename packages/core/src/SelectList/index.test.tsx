import React from 'react'
import 'jest-styled-components'

import { SelectList, ISelectListItemOption } from '.'
import { TestRender } from '../TestUtils'

const options: ReadonlyArray<ISelectListItemOption> = [
  { value: 'test1', label: 'test1' },
  { value: 'test2', label: 'test2', disabled: true },
  { value: 'test3', label: 'test3' },
]

const onSelect = () => {
  /** */
}

describe('SelectList', () => {
  test('SelectList', () => {
    const test1 = TestRender(
      <SelectList value="test1" options={options} onSelect={onSelect} />
    )
    expect(test1).toMatchSnapshot()

    const test2 = TestRender(
      <SelectList
        value="test1"
        options={options}
        onSelect={onSelect}
        maxHeight={200}
      />
    )
    expect(test2).toMatchSnapshot()
  })
})
