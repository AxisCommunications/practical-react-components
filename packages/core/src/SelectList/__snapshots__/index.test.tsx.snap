// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectList SelectList 1`] = `
.c4 {
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  font-size: 13px;
  line-height: 22px;
}

.c3 {
  height: 32px;
  padding: 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: rgb(255,255,255);
  background-color: rgb(103,58,183);
  pointer-events: none;
}

.c5 {
  height: 32px;
  padding: 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: rgb(102,102,102);
  pointer-events: none;
}

.c6 {
  height: 32px;
  padding: 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c6:hover {
  background-color: rgb(245,245,245);
}

.c2 {
  padding: 0;
  outline: none;
  list-style: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgb(255,255,255);
  color: rgb(41,41,41);
  min-width: 100%;
  border-radius: 2px;
  display: block;
  max-height: 320px;
  overflow: hidden;
  overflow-y: auto;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <ul
        className="c2"
        onKeyUp={[Function]}
        tabIndex={0}
      >
        <li
          className="c3"
          disabled={false}
          onClick={[Function]}
          role="menuitem"
          selected={true}
          value="test1"
        >
          <p
            className="c4"
          >
            test1
          </p>
        </li>
        <li
          className="c5"
          disabled={true}
          onClick={[Function]}
          role="menuitem"
          selected={false}
          value="test2"
        >
          <p
            className="c4"
          >
            test2
          </p>
        </li>
        <li
          className="c6"
          disabled={false}
          onClick={[Function]}
          role="menuitem"
          selected={false}
          value="test3"
        >
          <p
            className="c4"
          >
            test3
          </p>
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-2"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`SelectList SelectList 2`] = `
.c4 {
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  font-size: 13px;
  line-height: 22px;
}

.c3 {
  height: 32px;
  padding: 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: rgb(255,255,255);
  background-color: rgb(103,58,183);
  pointer-events: none;
}

.c5 {
  height: 32px;
  padding: 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: rgb(102,102,102);
  pointer-events: none;
}

.c6 {
  height: 32px;
  padding: 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c6:hover {
  background-color: rgb(245,245,245);
}

.c2 {
  padding: 0;
  outline: none;
  list-style: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgb(255,255,255);
  color: rgb(41,41,41);
  min-width: 100%;
  border-radius: 2px;
  display: block;
  max-height: 200px;
  overflow: hidden;
  overflow-y: auto;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <ul
        className="c2"
        onKeyUp={[Function]}
        tabIndex={0}
      >
        <li
          className="c3"
          disabled={false}
          onClick={[Function]}
          role="menuitem"
          selected={true}
          value="test1"
        >
          <p
            className="c4"
          >
            test1
          </p>
        </li>
        <li
          className="c5"
          disabled={true}
          onClick={[Function]}
          role="menuitem"
          selected={false}
          value="test2"
        >
          <p
            className="c4"
          >
            test2
          </p>
        </li>
        <li
          className="c6"
          disabled={false}
          onClick={[Function]}
          role="menuitem"
          selected={false}
          value="test3"
        >
          <p
            className="c4"
          >
            test3
          </p>
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-4"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;
